Documentation/internals/security-identities.rst

```diff
@@ -0,0 +1,48 @@
+.. only:: not (epub or latex or html)
+
+    WARNING: You are looking at unreleased Cilium documentation.
+    Please use the official rendered version released here:
+    https://docs.cilium.io
+
+.. _security_identities:
+
+*******************
+Security Identities
+*******************
+
+Security identities are generated from labels. They are stored as ``uint16``,
+which means the maximum limit for a security identity is ``2^16 - 1``. The
+minimum security identity is ``256``, due to the reserved identity range being
+from ``1`` to ``255``.
+
+There are two caveats to the above however:
+
+1) ClusterMesh
+2) Identities generated from CIDR-based policies
```

Comment:
I find this explanation a little bit confusing, because it attempts to describe what the ranges may denote, but on the way towards explaining all of the identity space, small assertions are made which are later contradicted.

For instance, the text ```The minimum security identity is ``256`` ``` suggests that the allocation range for cluster-wide identities starts from 256. But the text itself says that the minimum security identity is 256, and then the very next sentence describes that there are security identities below this minimum.

The "there are two caveats" part also contradicts the `uint16` storage assertion - if some of the system supports 32-bit identities then we can't outright declare that the security identity is a `uint16`, because some security identities are not uint16s.

I think that it's fairly safe to say that any arbitrary security identity will fall under the range `1 - 2^32-1`. There are limitations to this, for instance in general any identity that may be shared across multiple nodes (either inside the cluster or between Cilium-managed clusters) must fit inside a VXLAN or Geneve tunnel endpoint header's virtual network field, which is 24 bits for both of these protocols. So in practice, when handing traffic between nodes, the value must fall within `1 - 2^24-1`. In practice, we reserve the first 256 values and hence there may be additional special semantics for these identities. Then values above `2^24` cannot be shared directly between nodes via tunnel headers, so we avoid using these for consistent global identities; this is why the "local identities" set the lowest bit of the fourth byte.